### PR TITLE
feat:added Tag entity + CRUD + REST

### PR DESCRIPTION
**Pull request by VladislavPokhyl**

### CODE REVIEW

File: src/models/Tag.model.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { model, Schema, Document } from "mongoose";
+
+export interface Tag extends Document {
+  name: string;
+  stats: {
+    tweets: number;
+  };
+  createdAt: string;
```
Comment (VitaliiDenysiuk) [inline src/models/Tag.model.ts:9]:
```suggestion
  createdAt?: string;
```
filled by default

File: src/repositories/tags.repository.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { ObjectId } from "mongoose";
+
+import { Tag, TagModel } from "../models/Tag.model";
+
+export const getById = async (tagId: string | ObjectId) => {
```
Comment (VitaliiDenysiuk) [inline src/repositories/tags.repository.ts]:
```suggestion
export const getById = async (tagId: string) => {
```

File: src/repositories/tags.repository.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  return TagModel.findById(tagId);
+};
+
+export const create = async (tagData: Partial<Tag>) => {
```
Comment (VitaliiDenysiuk) [inline src/repositories/tags.repository.ts]:
```suggestion
export const create = (tagData: Partial<Tag>) => {
```
not await - async  was not needed

File: src/repositories/tags.repository.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  return TagModel.create(tagData);
+};
+
+export const getList = async () => {
```
Comment (VitaliiDenysiuk) [inline src/repositories/tags.repository.ts]:
```suggestion
export const getList =  () => {
```

File: src/repositories/tags.repository.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  return TagModel.find();
+};
+
+export const update = async (
+  tagId: string | ObjectId,
```
Comment (VitaliiDenysiuk) [inline src/repositories/tags.repository.ts]:
```suggestion
export const update =  (
  tagId: string ,
```

File: src/repositories/tags.repository.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  newTagData: Partial<Tag>
+) => {
+  return TagModel.findByIdAndUpdate(
+    tagId,
+    { ...newTagData, updatedAt: new Date().toISOString() },
+    { new: true }
+  );
+};
+
+export const deleteById = async (tagId: string | ObjectId) => {
```
Comment (VitaliiDenysiuk) [inline src/repositories/tags.repository.ts]:
```suggestion
export const deleteById = (tagId: string) => {
```

File: src/repositories/tags.repository.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import { ObjectId } from "mongoose";
```
Comment (VitaliiDenysiuk) [inline src/repositories/tags.repository.ts]:
```suggestion

```

File: src/services/tags.service.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import { ObjectId } from "mongoose";
```
Comment (VitaliiDenysiuk) [inline src/services/tags.service.ts]:
```suggestion

```

File: src/services/tags.service.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import { Tag } from "../models/Tag.model";
+import {
+  create,
+  deleteById,
+  getList,
+  update,
+  getById,
+} from "../repositories/tags.repository";
+
+export const getTagsList = async (): Promise<Array<Tag>> => {
```
Comment (VitaliiDenysiuk) [inline src/services/tags.service.ts]:
```suggestion
export const getTagsList =  () => {
```

File: src/services/tags.service.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+  return getList();
+};
+
+export const getTagById = async (id: string | ObjectId): Promise<Tag> => {
```
Comment (VitaliiDenysiuk) [inline src/services/tags.service.ts]:
```suggestion
export const getTagById = (id: string ) => {
```

File: src/services/tags.service.ts [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+  return getById(id);
+};
+
+export const createTag = async (tagData: Partial<Tag>): Promise<Tag> => {
+  const createResult = await create(tagData);
+  return createResult;
+};
```
Comment (VitaliiDenysiuk) [inline src/services/tags.service.ts]:
```suggestion
export const createTag = (tagData: Partial<Tag>) => {
  return create(tagData);
};
```

File: src/services/tags.service.ts [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+export const updateTag = async (
+  id: ObjectId | string,
+  newData: Partial<Tag>
+): Promise<Tag> => {
+  return update(id, newData);
+};
```
Comment (VitaliiDenysiuk) [inline src/services/tags.service.ts]:
```suggestion
export const updateTag = (
  id: string,
  newData: Partial<Tag>
) => {
  return update(id, newData);
};
```

File: src/services/tags.service.ts [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+
+export const deleteTagById = async (id: ObjectId | string) => {
```
Comment (VitaliiDenysiuk) [inline src/services/tags.service.ts]:
```suggestion
export const deleteTagById = async (id: string) => {
```

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]
add pagintation to getList request

**Comment by VladislavPokhyl** [comment]
image

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**